File: A/AA.m
//
//  AA.m
//  A
//
//  Created by Ho, Derrick on 6/4/16.
//  Copyright © 2016 Ho, Derrick. All rights reserved.
//

#import "AA.h"
#import <A/A-Swift.h>

NSString *_DHLocalizedString(NSString *key, NSString *table, NSString *comment, Class sender);

NSString *DHLocalizedString(NSString *key, NSString *table, NSString *comment, NSObject *sender) {
	return NSLocalizedStringWithDefaultValue(key, table, [NSBundle mainBundle], _DHLocalizedString(key, table, comment, [sender class]), comment);
}

NSString *_DHLocalizedString(NSString *key, NSString *table, NSString *comment, Class sender) {
	NSLog(@"> %@", sender);
	return sender ? NSLocalizedStringWithDefaultValue(key, table, [NSBundle bundleForClass:sender], _DHLocalizedString(key, table, comment, [sender superclass]), comment) : @"";
}

@implementation AA
@synthesize sayIt = _sayIt;

- (void)AAA {
	NSLog(@"AAAA bemo!");
}
@end

File: A/AA.swift
//
//  AA.swift
//  A
//
//  Created by Derrick  Ho on 6/8/16.
//  Copyright © 2016 Ho, Derrick. All rights reserved.
//

//public protocol ei {
//	var sayIt: String { get }
//}

public extension NSObject {
	public subscript (key: String) -> AnyObject? {
		get {
			return valueForKey(key)
		}
		set {
			setValue(newValue, forKey: key)
		}
	}
	
}

extension AA {
	public func thisIsASwiftMethod() {
		print("This is a Swift method extending an objective-c class")
		print("kudos")
	}
	public override func viewDidAppear(animated: Bool) {
		super.viewDidAppear(animated)
		print("viewdidappear on the swift side of the moon")
	}
	
	public func sayIt() -> String? {
		return self["_sayIt"] as! String?
		//return valueForKey("_sayIt") as! String?
	}
	public func setSayIt(s: String?) {
		self["_sayIt"] = s
		//setValue(s, forKey: "_sayIt")
	}
}
